open-access-variant-filtering-wf/main.nf

```diff
@@ -0,0 +1,199 @@
+#!/usr/bin/env nextflow
+
+/*
+  Copyright (C) 2021,  OICR
+  
+  This program is free software: you can redistribute it and/or modify
+  it under the terms of the GNU Affero General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+  
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU Affero General Public License for more details.
+  
+  You should have received a copy of the GNU Affero General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+  Authors:
+    Linda Xiang
+*/
+
+nextflow.enable.dsl = 2
+name = 'open-access-variant-filtering'
+short_name = 'open-filter'
+version = '0.1.0'  // package version
+
+// universal params go here, change default value as needed
+params.container = ""
+params.container_registry = ""
+params.container_version = ""
+params.cpus = 1
+params.mem = 1  // GB
+params.publish_dir = ""  // set to empty string will disable publishDir
+
+// tool specific parmas go here, add / change as needed
+params.study_id = ""
+params.analysis_id = ""
+params.regions_file = "NO_FILE_regions"
+params.output_type = ""
+params.apply_filters = [
+    'CaVEMan': "PASS", 
+    'Pindel': "PASS", 
+    'GATK-Mutect2': "PASS"
+]
+params.include = [
+    'CaVEMan': "INFO/CLPM=0 && INFO/ASRD>=0.93", 
+    'Pindel': "", 
+    'GATK-Mutect2': ""
+]
+params.exclude = [
+    'CaVEMan': "", 
+    'Pindel': "", 
+    'GATK-Mutect2': ""
+]
+params.open = true
+
+// if provided local files will be used
+params.analysis_metadata = "NO_FILE_metadata"
+params.vcf_file = "NO_FILE_vcf"
+
+params.api_token = ""
+params.song_url = ""
+params.score_url = ""
+params.cleanup = true
+
+params.download = [:]
+params.upload = [:]
+params.filter = [:]
+params.payloadGenVcf = [:]
+
+download_params = [
+    'song_cpus': params.cpus,
+    'song_mem': params.mem,
+    'score_cpus': params.cpus,
+    'score_mem': params.mem,
+    'song_url': params.song_url,
+    'score_url': params.score_url,
+    'api_token': params.api_token,
+    'publish_dir': params.publish_dir,
+    *:(params.download ?: [:])
+]
+
+upload_params = [
+    'cpus': params.cpus,
+    'mem': params.mem,
+    'song_url': params.song_url,
+    'score_url': params.score_url,
+    'api_token': params.api_token,
+    'publish_dir': params.publish_dir,
+    *:(params.upload ?: [:])
+]
+
+filter_params = [
+    'cpus': params.cpus,
+    'mem': params.mem,
+    'publish_dir': params.publish_dir,
```

Comment:
at workflow level, we don't always want to pass the workflow's publish_dir into processes. In this workflow, it makes more sense to disable publish_dir for the `filter` step, but enable it for payloadGen step. The reason is simple, output by `filter` is temporary while payloadGen step outputs the desired final version of the output.

So, here, it could set to `'publish_dir': ''`